src/vs/workbench/contrib/chat/browser/contrib/chatDynamicVariables.ts

```diff
@@ -41,12 +45,18 @@ export class ChatDynamicVariableModel extends Disposable implements IChatWidgetC
 	constructor(
 		private readonly widget: IChatWidget,
 		@ILabelService private readonly labelService: ILabelService,
+		@IFileService private readonly fileService: IFileService,
+		@IConfigurationService private readonly configService: IConfigurationService,
 	) {
 		super();
 		this._register(widget.inputEditor.onDidChangeModelContent(e => {
 			e.changes.forEach(c => {
 				// Don't mutate entries in _variables, since they will be returned from the getter
-				this._variables = coalesce(this._variables.map(ref => {
+				this._variables = coalesce(this._variables.map((ref) => {
+					if (c.text === `#file:${ref.filenameWithReferences}`) {
```

Comment:
That's the only way I made it work and its not ideal, any better ideas? cc @roblourens @joyceerhl

Reply:
What is this trying to do? Is this triggered when that text is inserted all at once?

Reply:
When we add the `(+N more)` label to a filename after nested references are resolved, this is run and as the result the variables are removed. The logic below infers that the variables text have changed and are hence the variables are no longer valid and need to be removed.

Reply:
Because we update the variable objects when we update the labels, this `if` check works, - if the variable text is really the same as the variable should have, do nothing.

Reply:
Can you set a flag during the time that you are making that edit, which says to ignore any `onDidChangeModelContent` that changes the variable text? Assuming this whole thing is sync.

Otherwise, I guess this is ok, but we could figure out how to make it more obvious what it's doing

Reply:
Yeah I don't think they are in sync - we can set the flag but cannot guarantee that the change event fired will be the in the queue 🤷

Reply:
This seems to break the logic that lets us delete one of these variables with one backspace, now the color goes away but I have to delete all the text manually

Reply:
Humm I don't think it ever worked for me like that, let me check 🔍